nft_ingester/src/bin/backfill/main.rs

```diff
@@ -197,82 +198,85 @@ async fn main() {
         let progress_bar = progress_bar.clone();
         let slots_processed = slots_processed.clone();
         let rate = rate.clone();
-        let shutdown_token = shutdown_token.clone();
 
         let slot_receiver = slot_receiver.clone();
 
-        let handle = tokio::spawn(async move {
-            while let Ok((slot, raw_block_data)) = slot_receiver.recv().await {
-                if shutdown_token.is_cancelled() {
-                    break;
-                }
-
-                // Process the slot
-                let raw_block: RawBlock = match RawBlock::decode(&raw_block_data) {
-                    Ok(rb) => rb,
-                    Err(e) => {
-                        error!("Failed to decode the value for slot {}: {}", slot, e);
-                        continue;
-                    },
-                };
-
-                if let Err(e) = consumer.consume_block(slot, raw_block.block).await {
-                    error!("Error processing slot {}: {}", slot, e);
+        let handle = tokio::task::spawn({
+            let cancellation_token = cancellation_token.child_token();
+            async move {
+                while let Ok((slot, raw_block_data)) = slot_receiver.recv().await {
+                    if cancellation_token.is_cancelled() {
```

Comment:
During the body execution shutting down is impossible

Reply:
it's fine because we want to finish the work we started